python/ray/serve/config.py

```diff
@@ -17,86 +19,75 @@ def _callable_is_blocking(func_or_class):
         return not inspect.iscoroutinefunction(func_or_class.__call__)
 
 
-class BackendConfig:
-    def __init__(self, config_dict, accepts_batches=False, is_blocking=True):
-        assert isinstance(config_dict, dict)
-        # Make a copy so that we don't modify the input dict.
-        config_dict = config_dict.copy()
+class BackendMetadata(BaseModel):
+    accepts_batches: bool = False
+    is_blocking: bool = True
+    autoscaling_config: Optional[Dict[str, Any]] = None
 
-        self.accepts_batches = accepts_batches
-        self.is_blocking = is_blocking
-        self.num_replicas = config_dict.pop("num_replicas", 1)
-        self.max_batch_size = config_dict.pop("max_batch_size", None)
-        self.batch_wait_timeout = config_dict.pop("batch_wait_timeout", 0)
-        self.max_concurrent_queries = config_dict.pop("max_concurrent_queries",
-                                                      None)
-        self.autoscaling_config = config_dict.pop("autoscaling", None)
 
-        if self.max_concurrent_queries is None:
+class BackendConfig(BaseModel):
+    """Configuration options for a backend, to be set by the user.
+    """
+
+    internal_metadata: BackendMetadata = BackendMetadata()
+    #: The number of worker processes to start up that will handle requests to this backend.  # noqa: E501
```

Comment:
I think you can make them multi-lines with proper spacing. I would highly prefer not to shutoff the linter in this case:
```diff
- #: The number of worker processes to start up that will handle requests to this backend.  # noqa: E501
+ #: The number of worker processes to start up that will 
+    handle requests to this backend. 
```

Reply:
Sounds good, looks like it works if you put `#:` on both lines